redis/src/cluster_client.rs

```diff
@@ -212,6 +222,13 @@ impl ClusterClientBuilder {
                 )));
             }
 
+            if client_name.is_some() && node.redis.client_name != *client_name {
```

Comment:
IMO we should return an error if `node.redis.client_name` is set, regardless of `client_name` it should be an error, because it might confuse users - they'll think that setting the client name on the initial nodes will affect the whole cluster, but it won't survive any change to the cluster. WDYT?

I see that the same mistake is made with the username and password, so this can be fixed in a separate PR.